src/PythNetworkHTTPClient.ts

```diff
@@ -0,0 +1,153 @@
+            throw new Error('Got a price update for an unknown product. This is a bug in the library, please report it to the developers.')
+        }
+
+        const priceData = parsePriceData(account.data)     
+        this._product_price.set(product, priceData);
+    }    
+
+    private handleAccount(key: PublicKey, account: AccountInfo<Buffer>, productOnly: boolean) {
+        const base = parseBaseData(account.data)
+        // The pyth program owns accounts that don't contain pyth data, which we can safely ignore.
+        if (base) {
+            switch (AccountType[base.type]) {
+                case 'Mapping':
+                    // We can skip these because we're going to get every account owned by this program anyway.
+                    break;
+                case 'Product':
+                    this.handleProductAccount(key, account)
+                    break;
+                case 'Price':
+                    if (!productOnly) {
+                        this.handlePriceAccount(key, account)
+                    } else {
+                        this._price_queue.push({
+                            key: key,
+                            account: account
+                        });
+                    }
+                    break;
+                case 'Test':
+                    break;
+                default:
+                    throw new Error(`Unknown account type: ${base.type}. Try upgrading pyth-client.`)
+            }
+        }
+    }
+
+    public async refreshData() {
+        const require_https = this._cluster === 'mainnet-beta' ? true : false;
+        const current_connection = new Connection(clusterApiUrl(this._cluster, require_https));
+
+        const accounts = await current_connection.getProgramAccounts(this._program_key, this._commitment);
+        for(const account of accounts) {
+            this.handleAccount(account.pubkey, account.account, true)
+        }
+
+        for(const queued of this._price_queue) {
+            this.handleAccount(queued.key, queued.account, false)
+        }
+    }
+
+    public assetsTypesKeys(): string[] {
+        return Array.from(this._asset_types);
+    }
+
+    public productsSymbolsKeys(): string[] {
+        return Array.from(this._product_symbols);
+    }
+
+    public producsWithAssetType(asset_type: string): Product[] {
+        if(!this._asset_types.has(asset_type))
+            return [];
+        
+        const result: Product[] = [];
+        this._products.forEach(current_product => {
+            if(current_product.asset_type === asset_type) {
+                result.push(current_product)
+            }
+        });
+
+        return result;
+    }
+
+    public producsWithSymbol(product_symbol: string): Product[] {
+        if(!this._product_symbols.has(product_symbol)) {
+            return [];
+        }
+        
+        const result: Product[] = [];
+        this._products.forEach(current_product => {
```

Comment:
I think you'd be better off using a Map for _product_symbols instead of a Set so that you don't have to iterate here. There will be at most a single Product with a given symbol.

Reply:
I've removed filter methods from the class